src/services/db/RepoService.ts

```diff
@@ -172,6 +224,67 @@ export default class RepoService extends GitHubService {
     })
   }
 
+  async readMediaDirectory(
+    sessionData: UserWithSiteSessionData,
+    readMediaInput: ReadMediaDirectoryInput
+  ): Promise<ReadMediaDirectoryOutput[]> {
+    const { directoryName } = readMediaInput.directoryInfo
+
+    logger.debug(`Reading media directory: ${directoryName}`)
+    const { siteName } = sessionData
+
+    // making as any to support incoming data from MediaDirectoryService
+    let filteredResult: any[] = []
+
+    const filterLogic = (file: any) =>
+      (file.type === "file" || file.type === "dir") &&
+      file.name !== PLACEHOLDER_FILE_NAME
+
+    if (this.isRepoWhitelisted(siteName)) {
+      const result = await this.gitFileSystemService.listDirectoryContents(
+        siteName,
+        directoryName
+      )
+
+      if (result.isErr()) {
+        throw result.error
+      }
+
+      filteredResult = result.value.filter(filterLogic)
+    } else if (readMediaInput.readFromGithub) {
+      filteredResult = readMediaInput.directoryInfo.files.filter(filterLogic)
+    }
+
+    const response = Promise.all(
+      filteredResult.map((curr) => {
+        if (curr.type === "dir") {
+          return {
+            name: curr.name,
+            type: curr.type,
+          }
+        }
+
+        if (this.isRepoWhitelisted(siteName)) {
+          return this.readMediaFile(sessionData, {
+            fileName: curr.name,
+            directoryName,
+          })
+        }
+        if (readMediaInput.readFromGithub) {
+          return getMediaFileInfo({
+            file: curr,
+            siteName,
+            directoryName,
+            mediaType: readMediaInput.directoryInfo.mediaType as MediaType,
+            isPrivate: readMediaInput.directoryInfo.isPrivate,
+          })
+        }
```

Comment:
is it possible that all three `if` conditions are not true, and the return type here would be `undefined`, which would not match the promise return type `ReadMediaDirectoryOutput`?